### PR TITLE
Add ingress boskos projects

### PR DESCRIPTION
**Pull request by MrHohn**
This PR makes all ingress jobs pull from ingress-project pool (projects have been created).

We will also need to change the old project reference here:
https://github.com/kubernetes/kubernetes/blob/2941c4bcbcabcc1516d86fa0e3345ee8262d21fa/test/e2e/framework/nodes_util.go#L79

/assign @rramkumar1 @krzyzacy

### CODE REVIEW

File: jobs/config.json [hunk 1]
```diff
@@ -219,9 +219,18 @@
   },
   "ci-ingress-gce-upgrade-e2e": {
     "args": [
-      "true"
+      "--check-leaked-resources",
+      "--cluster=",
+      "--extract=ci/latest",
+      "--extract=ci/latest",
```
Comment (krzyzacy) [inline jobs/config.json]: I doubt this will work... and probably the upgrade job change can be in a separate PR?

File: jobs/config.json [hunk 2]
```diff
@@ -1048,6 +1060,7 @@
       "--env=CLOUDSDK_CORE_PRINT_UNHANDLED_TRACEBACKS=1",
       "--extract=ci/k8s-stable1",
       "--gcp-node-image=gci",
+      "--gcp-project-type=ingress-project",
```
Comment (krzyzacy) [inline jobs/config.json]: `     "_comment": "AUTO-GENERATED by experiment/generate_tests.py - DO NOT EDIT.",`

I guess add the flag to https://github.com/kubernetes/test-infra/blob/master/experiment/test_config.yaml#L944-L947 and run `bazel run //experiment:generate_tests -- --yaml-config-path=experiment/test_config.yaml --json-config-path=jobs/config.json  --prow-config-path=prow/config.yaml && bazel run //jobs:config_sort`

Comment (MrHohn) [reply jobs/config.json]: Thanks for the command, updated now :)

File: experiment/test_config.yaml [hunk 3]
```diff
@@ -1019,23 +1020,23 @@ nodeK8sVersions:
   dev:
     args:
     - --repo=k8s.io/kubernetes=master
-    prowImage: gcr.io/k8s-testimages/kubekins-e2e:v20171220-29cb083c9-master
+    prowImage: gcr.io/k8s-testimages/kubekins-e2e:v20180123-1260ba9fc-master
   beta:
     args:
     - --repo=k8s.io/kubernetes=release-1.9
     prowImage: gcr.io/k8s-testimages/kubekins-e2e:v20171220-29cb083c9-1.9
```
Comment (krzyzacy) [inline experiment/test_config.yaml]: :joy: our script is obviously not super up-do-date thus ignored 1.9 stuff

**Comment by krzyzacy** [comment]
wanna maybe add a unit-test ensure all ingress job is using ingress project pool?

**Comment by MrHohn** [comment]
> wanna maybe add a unit-test ensure all ingress job is using ingress project pool?

Sounds good, not sure how to test though, would be great to have your advices :)

**Comment by MrHohn** [comment]
Fixes https://github.com/kubernetes/test-infra/issues/6284.

**Comment by krzyzacy** [comment]
maybe add a function to mess up more with https://github.com/kubernetes/test-infra/blob/master/jobs/config_test.py, inside `test_valid_job_config_json` maybe?

**Comment by MrHohn** [comment]
> wanna maybe add a unit-test ensure all ingress job is using ingress project pool?

@krzyzacy  Actually, would like to do this in follow-ups given there are more things to think about...Things like:
- Only gce ingress jobs will require this, but I haven't thought of a consistent way to know which job is a gce ingress job.
- Some of the gce ingress jobs don't need this (and can't do this), for instance "ingress image push" job need another dedicated project.
- Maybe more...

**Comment by krzyzacy** [comment]
sure
/lgtm
and :eyes:

**Comment by MrHohn** [comment]
> and :eyes:

Thanks! Sure, will watch them closely.

**Comment by MrHohn** [comment]
Humm, some ingress jobs start failing:
- https://k8s-testgrid.appspot.com/sig-network-gce#ingress-gce-e2e
- https://k8s-testgrid.appspot.com/sig-network-gke#gci-gke-ingress-release-1.9

**Comment by rramkumar1** [comment]
@MrHohn I think I know why ci-ingress-gce-e2e is failing. I think the job isn't able to pull the image from the gcr repo due to some permission issue. I will investigate why.

Not sure about the other one.

**Comment by krzyzacy** [comment]
@MrHohn the gke release branch ingress job is running against test endpoint...

**Comment by MrHohn** [comment]
Thanks for the fix! Just noticed ingress image push is failing as well :)
- http://k8s-testgrid.appspot.com/sig-network-gce#ingress-gce-image-push

**Comment by MrHohn** [comment]
> Just noticed ingress image push is failing as well :)

@rramkumar1 Will sent a fix soon.

**Comment by MrHohn** [comment]
Just for the record, thanks to @rramkumar1 most of the ingress jobs are passing now (turned out we also need to enable the test API). One or two of them are still failing but is irrelevant to our change.
